great_expectations/core/expectation_suite.py

```diff
@@ -1013,10 +1009,11 @@ def clean_empty(self, data):
     @pre_dump
     def prepare_dump(self, data, **kwargs):
         data = deepcopy(data)
-        if isinstance(data, ExpectationSuite):
-            data.meta = convert_to_json_serializable(data.meta)
-        elif isinstance(data, dict):
-            data["meta"] = convert_to_json_serializable(data.get("meta"))
+        for key in data:
+            if key.startswith("_"):
+                continue
+            data[key] = convert_to_json_serializable(data[key])
+
```

Comment:
we only specified "meta" here but non-serializable objects can live in other parts of the suite - this is the only minor change we need to make to completely remove `marshmallow__shade`